monkey/monkey_island/cc/repository/file_agent_log_repository.py

```diff
@@ -0,0 +1,35 @@
+import io
+
+from monkey_island.cc import repository
+from monkey_island.cc.models import AgentID
+from monkey_island.cc.repository import (
+    IAgentLogRepository,
+    IFileRepository,
+    RetrievalError,
+    UnknownRecordError,
+)
+
+AGENT_LOG_FILE_NAME_PREFIX = "agent_log_"
+
+
+class FileAgentLogRepository(IAgentLogRepository):
+    def __init__(self, file_repository: IFileRepository):
+        self._file_repository = file_repository
+
+    def upsert_agent_log(self, agent_id: AgentID, log_contents: str):
+        self._file_repository.save_file(
+            f"{AGENT_LOG_FILE_NAME_PREFIX}{agent_id}", io.BytesIO(log_contents.encode())
+        )
+
+    def get_agent_log(self, agent_id: AgentID) -> str:
+        try:
+            with self._file_repository.open_file(f"{AGENT_LOG_FILE_NAME_PREFIX}{agent_id}") as f:
+                log_contents = f.read().decode()
+                return log_contents
+        except repository.FileNotFoundError as err:
```

Comment:
The file repository already handles errors like this. Why doesn't it raise `UnknownRecordError` itself?

Reply:
I'm not sure. I guess we could completely remove `repository.FileNotFoundError` and raise `UnknownRecordError` everywhere instead. @mssalvatore?

Reply:
`UnknownRecordError` was added after `FileNotFoundError`. They both effectively serve the same purpose, so, good find.

The quick fix is to make `FileNotFoundError` inherit from `UnknownRecordError` instead of `RetrievalError`. Then we don't need to modify anything that uses `IFileRepository` right now. You can drop a `# TODO` in to remove `FileNotFoundError` at a later date.

If you feel strongly, you can remove it now, but do it in a separate PR.